src/nwb_benchmarks/benchmarks/time_remote_file_reading.py

```diff
@@ -113,9 +132,95 @@ def time_read_hdf5_nwbfile_ros3(self, s3_url: str):
         self.nwbfile, self.io, _ = read_hdf5_nwbfile_ros3(s3_url=s3_url, retry=False)
 
 
+class LindiFileReadLocalReferenceFileSystemBenchmark(BaseBenchmark):
+    """
+    Time the read of the Lindi HDF5 files with `pynwb` assuming that a local copy of the lindi
+    filesystem is available locally.
+    """
+
+    rounds = 1
+    repeat = 3
+    parameter_cases = lindi_hdf5_parameter_cases
+
+    def setup(self, s3_url: str):
+        """Create the local JSON LINDI reference filesystem if it does not exist"""
+        self.lindi_file = os.path.basename(s3_url) + ".lindi.json"
+        if not os.path.exists(self.lindi_file):
+            create_lindi_reference_file_system(s3_url=s3_url, outfile_path=self.lindi_file)
+
+    def time_read_lindi_nwbfile(self, s3_url: str):
+        """Read the NWB file with pynwb using LINDI with the local reference filesystem JSON"""
+        self.nwbfile, self.io, self.client = read_hdf5_nwbfile_lindi(rfs=self.lindi_file)
+
+    def time_read_lindi_jsonrfs(self, s3_url: str):
+        """Read the NWB file with LINDI directly using the local reference filesystem JSON"""
+        self.client = read_hdf5_lindi(rfs=self.lindi_file)
+
+
+class NWBLindiFileCreateLocalReferenceFileSystemBenchmark(BaseBenchmark):
+    """
+    Time the creation of a local Lindi JSON reference filesystem for a remote NWB file
+    as well as reading the NWB file with PyNWB when the local reference filesystem does not
+    yet exist.
+    """
+
+    rounds = 1
+    repeat = 3
+    parameter_cases = lindi_hdf5_parameter_cases
+
+    def setup(self, s3_url: str):
+        """Clear the LINDI JSON if it still exists"""
+        self.lindi_file = os.path.basename(s3_url) + ".lindi.json"
+        self.teardown(s3_url=s3_url)
+
+    def teardown(self, s3_url: str):
```

Comment:
```suggestion
        self.teardown()

    def teardown(self):
```

Reply:
Not sure why the s3_url needs is passed.

Reply:
I removed this

Reply:
Note, `setup` and `teardown` are reserved methods from `asv` and according to the docs, the `setup` and `teardown` are parametrized with the same parameters as the tests. See https://asv.readthedocs.io/en/stable/writing_benchmarks.html#parameterized-benchmarks

Reply:
Interesting. Good to know!